lib/improver/tests/blending/weighted_blend/test_WeightedBlendAcrossWholeDimension.py

```diff
@@ -713,6 +692,22 @@ def test_specific_cycletime(self):
         self.assertEqual(result.coord('time').points,
                          cube.coord('time').points)
 
+    @ManageWarnings(
+        ignored_messages=["Collapsing a non-contiguous coordinate"])
+    def test_attributes_dict(self):
+        """Test updates to attributes on output cube"""
+        attributes_dict = {"source": "IMPROVER", "history": "cycle blended"}
+        for key in self.cube.attributes:
+            if "mosg__" in key:
+                attributes_dict[key] = "remove"
+        expected_attributes = {"source": "IMPROVER",
+                               "history": "cycle blended",
+                               "title": self.cube.attributes["title"]}
+        coord = "forecast_reference_time"
+        plugin = WeightedBlendAcrossWholeDimension(coord)
+        result = plugin.process(self.cube, attributes_dict=attributes_dict)
+        self.assertDictEqual(result.attributes, expected_attributes)
+
```

Comment:
Please could you add a test that modifies only a subset of the attributes. I can only see tests for none and all.

Reply:
This test updates some attributes, removes others, and leaves "title" unchanged.  This is indicated by having `self.cube.attributes["title"]` as the expected value of the title attribute on the output cube.